connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/SparkConnectPlanExecution.scala

```diff
@@ -163,7 +165,7 @@ private[execution] class SparkConnectPlanExecution(executeHolder: ExecuteHolder)
               // Collect errors and propagate them to the main thread.
               .andThen {
                 case Success(_) =>
-                  executePlan.eventsManager.postFinished()
+                  executePlan.eventsManager.postFinished(Some(totalNumRows))
```

Comment:
I see test failed
```
SPARK-41224: collect data using arrow *** FAILED *** (290 milliseconds)
[info]   Some(50) did not equal Some(100) (SparkConnectServiceSuite.scala:905)
[info]   Analysis:
[info]   Some(value: 50 -> 100)
```
We might have to move this `postFinished` after `ThreadUtils.awayReady`

Reply:
> I believe we have to move postFinished after ThreadUtils.awayReady

+1, becaues resultHandler only sends signal to the loop below that sends batch